### ISSUE TITLE
Monolog handler is not compatible with monolog v3

### ISSUE THREAD

**Issue by W0rma** [getsentry/sentry-php#1318]
[`Sentry\Monolog\Handler`](https://github.com/getsentry/sentry-php/blob/master/src/Monolog/Handler.php) extends [`Monolog\Handler\AbstractProcessingHandler`](https://github.com/Seldaek/monolog/blob/main/src/Monolog/Handler/AbstractProcessingHandler.php). 

The signature of the constructor has changed in [monolog 3](https://github.com/Seldaek/monolog/blob/3.0.0/UPGRADE.md#300) which leads to a fatal error:

```
Fatal error: Declaration of Sentry\Monolog\Handler::write(array $record): void must be compatible with Monolog\Handler\AbstractProcessingHandler::write(Monolog\LogRecord $record): void in /var/www/html/vendor/sentry/sentry/src/Monolog/Handler.php on line 51
```

**Comment by Jean85** [comment]
This is definitely something that we need to tackle, but I'm unsure how we should proceed: do we do a conditional declaration with an `if (/** detect monolog version **/)` at the top? or do we add a new, separate `HandlerV3`?

WDYT @ste93cry @stayallive? Any possible outside impact that you may think of?

**Comment by simPod** [comment]
We should mind this line https://github.com/getsentry/sentry-php/blob/6e8d3e24a7045b7222f8ad7987b7ebc6f0faf394/src/Monolog/Handler.php#L68 where it now reads only 'context' but it should support 'extra' as well. See https://github.com/Seldaek/monolog/blob/main/doc/message-structure.md

**Comment by jiripudil** [comment]
Quoting the [release notes](https://github.com/Seldaek/monolog/releases/tag/3.0.0),

> If you want to support multiple Monolog versions this should be possible by type-hinting nothing, or `array|LogRecord` if you support PHP 8.0+. You can then code against the $record using Monolog 2 style as LogRecord implements ArrayAccess for BC.

It seems that the BC code in `LogRecord` even handles [things such as `level`](https://github.com/Seldaek/monolog/blob/60ad5183b5e5d6c9d4047e9f3072d36071dcc161/src/Monolog/LogRecord.php#L79-L88) in a way that it returns the same integer value as in 2.0. So perhaps this is simply a matter of dropping the `array` type-hint – or widening it to `array|LogRecord` if you can afford to bump the minimum PHP version to 8.0.

**Comment by ste93cry** [comment]
>Any possible outside impact that you may think of?

No, I think that either solutions are fine. @W0rma already proposed an implementation in the referenced PR using the conditional declaration though

**Comment by gjuric** [comment]
Can we expect a new release with this PR merged soon?

**Comment by chekalsky** [comment]
Also interested in this release :)

**Comment by ste93cry** [comment]
On its way in the next days!

**Comment by Jean85** [comment]
Released with https://github.com/getsentry/sentry-php/releases/tag/3.6.0 :tada:
